llvm/lib/Transforms/InstCombine/InstCombineCalls.cpp

```diff
@@ -1174,6 +1174,163 @@ static Instruction *moveAddAfterMinMax(IntrinsicInst *II,
   return IsSigned ? BinaryOperator::CreateNSWAdd(NewMinMax, Add->getOperand(1))
                   : BinaryOperator::CreateNUWAdd(NewMinMax, Add->getOperand(1));
 }
+
+
+static bool rightDistributesOverLeft(Instruction::BinaryOps ROp, bool HasNUW,
```

Comment:
Please add some header comments for this function.

Reply:
Can you provide alive2 proof for this function (`(X LOp Y) ROp Z -> (X ROp Z) LOp (Y ROp Z)`)?